spec/requests/proceeding_types/searches_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'rails_helper'
+
+RSpec.describe 'ProceedingTypes/SearchController', type: :request do
+  before { seed_live_data }
+  describe 'POST proceeding_types/search' do
+    let(:headers) { { 'CONTENT_TYPE' => 'application/json' } }
+    let(:params) do
+      {
+        search_term: search_term
+      }
+    end
+    let(:search_term) { 'Occupation' }
+    subject { post proceeding_types_searches_path, params: params.to_json, headers: headers }
+
+    context 'when the params are valid' do
+      before { subject }
+      let(:expected_result) do
+        {
+          success: true,
+          data: [
+            {
+              meaning: 'Occupation order',
+              ccms_code: 'DA005',
+              description: 'to be represented on an application for an occupation order.',
+              ccms_category_law: 'Family',
+              ccms_matter: 'Domestic abuse'
+            }
+          ]
+        }
+      end
+
+      it 'returns a successful response' do
+        expect(response).to have_http_status(200)
+        expect(response.media_type).to eql('application/json')
+        expect(JSON.parse(response.body)).to match_json_expression(expected_result)
+      end
+    end
+
+    context 'when no matches found' do
+      before { subject }
+      let(:search_term) { 'nonexistant' }
+      let(:expected_result) do
+        {
+          success: false,
+          error: 'no matches found'
+        }
+      end
+
+      it 'returns an unsuccessful response' do
+        expect(response).to have_http_status(400)
+        expect(response.media_type).to eql('application/json')
+        expect(JSON.parse(response.body)).to match_json_expression(expected_result)
+      end
+    end
+
+    context 'when an error occurs' do
+      before do
+        allow(ProceedingTypeFullTextSearch).to receive(:call).and_raise(StandardError.new('Unexpected error in full text search'))
+        subject
+      end
+      let(:expected_result) do
+        {
+          success: false,
+          error: 'StandardError',
+          message: 'Unexpected error in full text search'
+        }
+      end
+
+      it 'returns an successful response' do
+        expect(response).to have_http_status(400)
```

Comment:
Probably a hangover from Apply, but shouldn't this be `it returns and unsuccessful response` ?